docs/04-api-reference/front-commerce-core/graphql.mdx

```diff
@@ -0,0 +1,435 @@
+
+See
+[`@front-commerce/remix/playground/yoga.ts`](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/packages/remix/playground/yoga.ts?ref_type=heads)
+for usage example.
+
+:::
+
+## `createGraphQLApi`
+
+Create a GraphQL method to execute GraphQL queries and mutations
+
+```ts
+const api = createGraphQLApi(request, user, config);
+```
+
+Arguments:
+
+- `request` (`Request`): The request object
+- `user` (`User`): The user context
+- `config` (`Config`): The application configuration
+
+_Example_
+
+```ts
+import { createGraphQLApi } from "@front-commerce/core/graphql";
+import { AcmeQuery } from "~/graphql/graphql";
+
+const AcmeGraphQLApi = createGraphQLApi(request, user, config);
+
+const response = await AcmeGraphQLApi(AcmeQuery);
+```
+
+## `makeErrorLoggerInterceptor`
+
+Intercept and log errors
+
+```js
+makeErrorLoggerInterceptor({
+  isMaintenance: maintenance,
+  ignoreForbiddenErrors: ignore,
+});
+```
+
+Arguments:
+
+- `maintenance` (`function`): a function that recieve error as parameter and
+  return a `boolean` to tell if website is in maintenance mode
+- `ignore` (`boolean`): a boolean to tell if forbidden errors should be ignored
+
+_Example_
+
+```ts
+import { makeErrorLoggerInterceptor } from "@front-commerce/core/graphql";
+
+const errorLoggerInterceptor = makeErrorLoggerInterceptor({
+  isMaintenance: (error) =>
+    error.response &&
+    (error.response.status === 503 ||
+      (error.response.status === 500 &&
+        /maintenance mode .* enabled/.test(error.response.data))),
+  ignoreForbiddenErrors: ignoreForbiddenErrors,
+});
+```
+
+## `limitRateByClientIp`
+
+Limit a GraphQL mutation by client IP
+
+```ts
+limitRateByClientIp(config, resolver);
+```
+
+Arguments:
+
+- `config` (`object`): rate limiter configuration object:
+  - `max` (`number`): number of calls allowed within window
+  - `window` (`number`): window of maximum calls
+- `resolver` (`function`): GraphQL resolver
+
+_Example_
+
+```ts title="extension/acme/graphql/modules/contact/resolver.ts"
+import { limitRateByClientIp } from "@front-commerce/core/graphql";
+
+export default {
+  Mutation: {
+    sendContactEmail: limitRateByClientIp(
+      { max: 5, window: "1m" },
+      (_, { template, data, storeEmail }, { loaders }) =>
+        loaders.Contact.sendContactEmail(template, data, storeEmail)
+    ),
+  },
+};
+```
+
+## `reorderForIds`
+
+Reorder an array by ids
+
+```ts
+reorderForIds(ids, (key = "id"));
+```
+
+Arguments:
+
+- `ids` (`string[]` default to `id`): ids to reorder
+- `key` (`string`): key to use to order
+
+_Example_
+
+```ts
+import { reorderForIds } from "@front-commerce/core/graphql";
+
+const orderer = reorderForIds([2, 3, 1], "id");
+const reordered = orderer([{ id: 1 }, { id: 2 }, { id: 3 }]);
+```
+
+## `reorderForIdsCaseInsensitive`
+
+Same as [`reorderForIds`](#) but case insensitive
+
+## `makeBatchLoaderFromSingleFetch`
+
+Convert a single fetch to a batch loader
```

Comment:
**thought(non-blocking):** Not sure were it'll end up, but an explanation (_or a link to one_) about what's a batch loader could be great here. This can be done later though.